tasks/actorstate/multisig.go

```diff
@@ -0,0 +1,157 @@
+package actorstate
+
+import (
+	"context"
+
+	"github.com/filecoin-project/lotus/chain/actors/builtin/multisig"
+	"github.com/filecoin-project/lotus/chain/types"
+	sa0builtin "github.com/filecoin-project/specs-actors/actors/builtin"
+	sa2builtin "github.com/filecoin-project/specs-actors/v2/actors/builtin"
+	"go.opentelemetry.io/otel/api/global"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/sentinel-visor/metrics"
+	"github.com/filecoin-project/sentinel-visor/model"
+	multisigmodel "github.com/filecoin-project/sentinel-visor/model/actors/multisig"
+)
+
+func init() {
+	Register(sa0builtin.MultisigActorCodeID, MultiSigActorExtractor{})
+	Register(sa2builtin.MultisigActorCodeID, MultiSigActorExtractor{})
+}
+
+type MultiSigActorExtractor struct{}
+
+func (m MultiSigActorExtractor) Extract(ctx context.Context, a ActorInfo, node ActorStateAPI) (model.Persistable, error) {
+	ctx, span := global.Tracer("").Start(ctx, "MultiSigActor")
+	defer span.End()
+
+	stop := metrics.Timer(ctx, metrics.ProcessingDuration)
+	defer stop()
+
+	ec, err := NewMultiSigExtractionContext(ctx, a, node)
+	if err != nil {
+		return nil, err
+	}
+
+	transactionModels, err := ExtractMultisigTransactions(a, ec)
+	if err != nil {
+		return nil, xerrors.Errorf("extracting multisig actor %s with head %s transactions: %w", a.Address, a.Actor.Head, err)
+	}
+	return &multisigmodel.MultisigTaskResult{TransactionModel: transactionModels}, nil
+}
+
+func ExtractMultisigTransactions(a ActorInfo, ec *MsigExtractionContext) (multisigmodel.MultisigTransactionList, error) {
+	var out multisigmodel.MultisigTransactionList
+	if ec.IsGenesis() {
+		if err := ec.CurrState.ForEachPendingTxn(func(id int64, txn multisig.Transaction) error {
+			// the ordering of this list must always be preserved as the 0th entry is the proposer.
+			approved := make([]string, len(txn.Approved))
+			for i, addr := range txn.Approved {
+				approved[i] = addr.String()
+			}
+			out = append(out, &multisigmodel.MultisigTransaction{
+				MultisigID:    a.Address.String(),
+				StateRoot:     ec.CurrTs.ParentState().String(),
+				Height:        int64(ec.CurrTs.Height()),
+				TransactionID: id,
+				To:            txn.To.String(),
+				Value:         txn.Value.String(),
+				Method:        uint64(txn.Method),
+				Params:        txn.Params,
+				Approved:      approved,
+			})
+			return nil
+		}); err != nil {
+			return nil, err
+		}
+		return out, nil
+	}
+
+	changes, err := multisig.DiffPendingTransactions(ec.PrevState, ec.CurrState)
```

Comment:
note: transactions that were removed could either have been canceled, executed and failed or executed and succeeded. Needing to spend a bit more time thinking if we can gather that info here, my gut says no..